src/list.rs

```diff
@@ -60,9 +79,9 @@ impl<T: DeserializeOwned> IntoIterator for List<T> {
 /// This struct is created by the `into_iter` method on `List`.
 ///
 /// Upon reaching the end of a page, further pages will be requested and the
-/// iterator will continue yielding items from those pages. Therefore the
-/// associated `Item` type must be a [`crate::Result`]`<T>` as this request can
-/// fail.
+/// iterator will continue yielding items from those pages. If one of the
+/// subsequent requests fails (which it shouldn't), the error is logged to the
+/// console and the iterator stops yielding items.
```

Comment:
Do you think it would be better to panic?

Reply:
Hmm.. this is quite tricky to fix :thinking: because we can't report errors in favor of being more ergonomic. I don't like it but leave it for now with a todo or something.

Reply:
My logic is that the requests are quite unlikely to fail if the first page succeeds. However, if you'd prefer, I'll change the Item type back to `crate::Result<T>` in another PR.